src/client/game.cpp

```diff
@@ -2060,15 +2060,17 @@ void Game::openInventory()
 	InventoryLocation inventoryloc;
 	inventoryloc.setCurrentPlayer();
 
-	if (!client->modsLoaded()
-			|| !client->getScript()->on_inventory_open(fs_src->m_client->getInventory(inventoryloc))) {
-		TextDest *txt_dst = new TextDestPlayerInventory(client);
-		auto *&formspec = m_game_ui->updateFormspec("");
-		GUIFormSpecMenu::create(formspec, client, m_rendering_engine->get_gui_env(),
-			&input->joystick, fs_src, txt_dst, client->getFormspecPrepend(), sound);
+	if (!client->modsLoaded() || !client->getScript()->on_inventory_open(fs_src->m_client->getInventory(inventoryloc)))
+    	return;
+	if (fs_src->getForm().empty())
+   		return;
```

Comment:
You must still free the memory in case it's not passed to `GUIFormSpecMenu::create`. Example:

```C++
std::unique_ptr<PlayerInventoryFormSource> fs_src(new PlayerInventoryFormSource(client));

// if (....)
//      return statements as-is. memory is freed automatically.

// else: pass fs_src.release()   to GUIFormSpecMenu::create
```

https://en.cppreference.com/w/cpp/memory/unique_ptr/release

Reply:
Whoops, sorry, I was fixing it and pushed it right as you commented. I had not forgotten about it which was why I didn't resolve sfan's comment. Is `delete fs_src` okay or should I switch it to `fs_src.release()`?

Reply:
Both types work. Normal pointers require manual cleanup, and smart pointers do it themselves, which helps to avoid memory leaks. The code will work either way.